lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1393,4 +1399,19 @@ public function __clone()
             $this->params[$name] = clone $param;
         }
     }
+
+    /**
+     * Filters out empty values from the given argument list.
+     * Fixes a BC break introduces in 2.11: https://github.com/doctrine/dbal/issues/4282
+     *
+     * @param array<mixed> $args
+     *
+     * @return array<mixed>
+     *
+     * @todo remove for 3.0
+     */
+    private static function filterEmptyArgs(array $args): array
+    {
+        return array_values(array_filter($args));
```

Comment:
Is `array_values()` necessary here? The array keys are not used anywhere explicitly, and PHP seems to ignore the keys of the array passed via `...`.

Reply:
Indeed! Because PHP errors when non-numeric keys are used, I was pretty sure that it would error if non-sequentials keys were used... I was wrong. Fixed.

The above question stands even more then: should we keep `filterEmptyArgs()` as documentation? Or should we use `array_filter()` inline? But we lose the link to the issue and the TODO.